htmd/parameterization/fftype.py

```diff
@@ -289,79 +278,31 @@ def test_mol(self):
 
         for name, method, chargetuple in toTest:
             with self.subTest(name=name, method=method, chargetuple=chargetuple):
+
                 refDir = os.path.join(self.refDir, name, method)
                 self._init_mol(name, method, chargetuple)
-                # self._generate_references(name, method)
-                with open(os.path.join(refDir, 'mol_props.yaml'), 'r') as infile:
-                    refProps = self.yamlload(infile)
+
+                with open(os.path.join(refDir, 'mol_props.yaml')) as infile:
+                    refProps = yamlload(infile)
                 self._test_mol_props(**refProps)
+
                 with open(os.path.join(refDir, 'params.p'), 'rb') as infile:
-                    refParams = self.pickleload(infile)
+                    refParams = pickleload(infile)
                 self._test_params(refParams)
-                with open(os.path.join(refDir, 'intermediary_files.yaml'), 'r') as infile:
-                    refFiles = self.yamlload(infile)
+
+                with open(os.path.join(refDir, 'intermediary_files.yaml')) as infile:
+                    refFiles = yamlload(infile)
                 self._test_intermediary_files(refFiles)
 
     def test_broken_atomname(self):
```

Comment:
`test_broken_name` fails with `RunTimeError`, but rather on calling. This was the important test that I took a long time to fix, for when a user passes garbage (in this case, a molecule with atomnames completely broken), the program exits reasonably (and with a nice message of what's going on). I am currently getting a parmchk2 segfault. Do you get the same? I thought my parmchk2 could be broken, but it parses the `test_mol` ok.

Reply:
Fixed!